burnMyDread/moby
# 39149 default capabilites

**- What I did**
I extended the capabilities work started in:
https://github.com/moby/moby/commit/80d7bfd54d19cc25f4855b4f2e9e4ad2e47f86c0
To allow default capabilities configuration for the docker daemon.

**- How I did it**
I added a configuration option to set the default capabilities. I reused the code introduced in:
https://github.com/moby/moby/commit/80d7bfd54d19cc25f4855b4f2e9e4ad2e47f86c0
To handle the logic to add the caps as needed. I added some logic in the deamon configuration to copy the capabilities settings if they are set in the daemon but not the container's host config.

**- How to verify it**
If you start the daemon with the capabilities flag and run:
`capsh --print`
In the container. You should just see the capabilities you configured. 
(I also updated the unit tests to use the code.)

**- Description for the changelog**

Added default capabilities configuration 

**- A picture of a cute animal (not mandatory but encouraged)**
![alt text](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSYd9Xzg3K2pgj_9cDoWAE7FM-KkUOTV8l4oJ83He-jx_IIHWPYEQ "Two for one")
